00X-ABI-Stable-Module-API.md

```diff
@@ -0,0 +1,179 @@
+| Title  | ABI Stable Module API                 |
+|--------|---------------------------------------|
+| Author | @mhdawson, @stefanbu, @Ianwjhalliday  |
+| Status | DRAFT                                 |
+| Date   | 2016-04-29                            |
+
+# Introduction
+
+The module ecosystem is one of the features making Node.js the fastest
+growing runtime.  An important subset of modules are those which have a
+native component. In specific cases it is important to be able to leverage native
+code in order to be able to:
+
+* use existing third party components written in C/C++
+* access physical devices, for example a serial port
+* expose functionality from the operating system not otherwise available
+* achieve higher speed than possible in native JavaScript
+
+There is an existing interface (addons) that allows you to build these modules
+as a dynamically-linked shared module that can be loaded into Node.js and
+used as an ordinary Node.js module (for more details see
+https://nodejs.org/api/addons.html)
+
+Unfortunately in the existing implementation the V8
+(the JavaScript runtime for Node.js) APIs are exposed to modules. In 
+addition Google changes these APIs on a regular basis. This results in 
+the following issues:
+
+* Native modules must be recompiled for each version of Node.js
+* The code within modules must often be modified for a new version of
```

Comment:
'must often' is too strong. Node.js 6.x didn't have API changes and didn't require module code to be modified. I would suggest wording this as 'The code within modules may need to be modified for a new version of Node.js'.